src/Symfony/Component/Intl/Data/Generator/TimezoneDataGenerator.php

```diff
@@ -167,15 +161,15 @@ private function generateZones(string $locale, ArrayAccessibleResourceBundle $ty
 
             return null;
         };
-        $regionFormat = $accessor(['zoneStrings', 'regionFormat']) ?? '{0}';
-        $fallbackFormat = $accessor(['zoneStrings', 'fallbackFormat']) ?? '{1} ({0})';
+        $regionFormat = $reader->readEntry($tempDir, $locale, ['zoneStrings', 'regionFormat']);
+        $fallbackFormat = $reader->readEntry($tempDir, $locale, ['zoneStrings', 'fallbackFormat']);
         $zoneToCountry = self::generateZoneToCountryMapping($windowsZonesBundle);
-        $resolveName = function (string $id, string $city = null) use ($locale, $regionFormat, $fallbackFormat, $zoneToCountry): string {
+        $resolveName = function (string $id, string $city = null) use ($reader, $tempDir, $locale, $regionFormat, $fallbackFormat, $zoneToCountry): ?string {
             if (isset($zoneToCountry[$id])) {
                 try {
-                    $country = $this->regionDataProvider->getName($zoneToCountry[$id], $locale);
+                    $country = $reader->readEntry($tempDir.'/region', $locale, ['Countries', $zoneToCountry[$id]]);
                 } catch (MissingResourceException $e) {
-                    $country = $this->regionDataProvider->getName($zoneToCountry[$id], 'en');
+                    return null;
```

Comment:
using the "english" country name as a fallback hardcoded is wrong, it should be determined during runtime fallback resolution. This is causing the removed timezone IDs below, but it's OK.